tests/infra/server.py

```diff
@@ -0,0 +1,88 @@
+import chalice.config
+import functools
+import logging
+import os
+import requests
+import threading
+import types
+
+from chalice.cli import CLIFactory
+from chalice.local import LocalDevServer, ChaliceRequestHandler
+
+import contextlib
+import socket
+
+
+def unused_tcp_port():
+    with contextlib.closing(socket.socket()) as sock:
+        sock.bind(('127.0.0.1', 0))
+        return sock.getsockname()[1]
+
+
+class SilentHandler(ChaliceRequestHandler):
+    """
+    The default Chalice request handler is very chatty.  We don't want that polluting our unit test output, so we
+    replace the `log_message` function with something quieter.
+    """
+    def log_message(self, *args, **kwargs):
+        pass
```

Comment:
I'm pretty sure this can be addressed by reconfiguring log levels on the relevant loggers, no?